boskos/metrics/metrics.go

```diff
@@ -99,19 +108,35 @@ func main() {
 	logrus.WithError(http.ListenAndServe(":8080", nil)).Fatal("ListenAndServe returned.")
 }
 
+func filterMetrics(src map[string]int) map[string]int {
+	metricStates := sets.NewString(states...)
+	dest := map[string]int{}
+	// Making sure all metrics are created
+	for state := range metricStates {
+		dest[state] = 0
+	}
+	dest[common.Other] = 0
+	for state, value := range src {
+		if state != common.Other && metricStates.Has(state) {
+			dest[state] = value
+		} else {
+			dest[common.Other] += value
```

Comment:
It isn't obvious to me why wouldn't want to do dest[state] += value even when state isn't a part of `--resource-state`? 

I'm sure there's a good reason but it isn't immediately obvious to me glancing through this review.

Reply:
too many negatives in your sentence :)
If someone is interested in metrics about a given states, it is easy to do, but I think it is also helpful to know what other states are not captured. The user can decide or not to query that metric. 

In Mason use cases, when we return a leased resource, we mark its status to be the name of the main resource. so you have the main resource A{state: Free} which depends on B{state: A} and C{state: A}. When you acquire A, you can use AcquireByState to get B and C. For monitoring, you don't care about the state A, but knowing that you have N number of resources that are ready to be used is a great information.
See https://velodrome.istio.io/dashboard/db/boskos-dashboard?orgId=1&from=now-6h&to=now